pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,471 @@
+}
+
+func updateTestCountsInSuite(suite *junit.TestSuite) {
+	var numTests, numFailed uint
+	for _, test := range suite.TestCases {
+		numTests++
+		if test.FailureOutput != nil {
+			numFailed++
+		}
+	}
+	for _, child := range suite.Children {
+		updateTestCountsInSuite(child)
+		numTests += child.NumTests
+		numFailed += child.NumFailed
+	}
+	suite.NumTests = numTests
+	suite.NumFailed = numFailed
+}
+
+// CheckTestCase returns a test case based on whether a test has passed certain criteria across job runs
+func (r minimumRequiredPassesTestCaseChecker) CheckTestCase(ctx context.Context, jobRunJunits map[jobrunaggregatorapi.JobRunInfo]*junit.TestSuites) *junit.TestSuite {
+	topSuite := &junit.TestSuite{
+		Name:      "minimum-required-passes-checker",
+		TestCases: []*junit.TestCase{},
+	}
+	bottomSuite := addToTestSuiteFromSuiteNames(topSuite, r.id.testSuites)
+
+	testName := fmt.Sprintf("test '%s' has required number of successful passes across payload jobs", r.id.testName)
+	if len(r.testNameSuffix) > 0 {
+		testName += fmt.Sprintf(" for %s", r.testNameSuffix)
+	}
+	testCase := &junit.TestCase{
+		Name: testName,
+	}
+	bottomSuite.TestCases = append(bottomSuite.TestCases, testCase)
+
+	start := time.Now()
+	successCount := 0
+	currDetails := &jobrunaggregatorlib.TestCaseDetails{
+		Name:          r.id.testName,
+		TestSuiteName: strings.Join(r.id.testSuites, jobrunaggregatorlib.TestSuitesSeparator),
+	}
+	for jobRun, testSuites := range jobRunJunits {
+		status := testSkipped
+		// Now go through test suites check test result
+		for _, testSuite := range testSuites.Suites {
+			status = getTestStatus(r.id, testSuite)
+			found := false
+			switch status {
+			case testPassed:
+				found = true
+				successCount++
+			case testFailed:
+				found = true
+			}
+			if found {
+				break
+			}
+		}
+		r.addTestResultToDetails(currDetails, jobRun, status)
+	}
+	currDetails.Summary = fmt.Sprintf("Total job runs: %d, passes: %d, failures: %d, skips %d", len(jobRunJunits), len(currDetails.Passes), len(currDetails.Failures), len(currDetails.Skips))
+	detailsYaml, err := yaml.Marshal(currDetails)
+	if err != nil {
+		return nil
+	}
+	testCase.Duration = time.Since(start).Seconds()
+	testCase.SystemOut = string(detailsYaml)
+	if successCount < r.requiredNumberOfPasses {
+		testCase.FailureOutput = &junit.FailureOutput{
+			Message: fmt.Sprintf("required minimum successful count %d, got %d", r.requiredNumberOfPasses, successCount),
+		}
+	}
+	updateTestCountsInSuite(topSuite)
+	return topSuite
+}
+
+// JobRunTestCaseAnalyzerOptions
+// 1. either gets a list of jobs from big query that meet the passed criteria: platform, network, or uses the passed jobs
+// 2. finds job runs for matching jobs for the specified payload tag
+// 3. runs all test case checkers and constructs a synthetic junit
+type JobRunTestCaseAnalyzerOptions struct {
+	payloadTag          string
+	workingDir          string
+	jobRunStartEstimate time.Time
+	clock               clock.Clock
```

Comment:
godoc as well.

Reply:
Turned out it is not needed. Inherited code.